src/test/java/com/techsophy/tsf/runtime/form/service/FormDataServiceElasticDisabledTest.java

```diff
@@ -202,6 +205,197 @@ void saveFormDataWithoutCollectionTest() throws IOException
+        formDataMap.put(UPDATED_BY_NAME,UPDATED_BY_USER_NAME);
+        Mockito.when(mockMongoTemplate.save(any(),anyString())).thenThrow(new RuntimeException("E11000 duplicate key"));
+        Mockito.when(mockObjectMapper.convertValue(any(),eq(FormDataDefinition.class))).thenReturn(new FormDataDefinition());
+        Assertions.assertThrows(InvalidInputException.class,()->mockFormDataServiceImpl.saveFormData(formDataSchemaTest));
+    }
+
+    @Test
+    void saveFormDataWithUniqueKeyCheckCollectionNotExist() throws IOException
+    {
+        Map<String, Object> testFormData = new HashMap<>();
+        testFormData.put(NAME, NAME_VALUE);
+        testFormData.put(AGE,AGE_VALUE);
+        Map<String, Object> testFormMetaData = new HashMap<>();
+        testFormMetaData.put(FORM_VERSION, 1);
+        FormDataSchema formDataSchemaTest=new FormDataSchema(null,TEST_FORM_ID,
+                TEST_VERSION,testFormData,testFormMetaData);
+        doReturn(userList).when(mockUserDetails).getUserDetails();
+//        Mockito.when(mockTokenUtils.getTokenFromContext()).thenReturn(TEST_TOKEN);
+        FormResponseSchema formResponseSchemaTest = new FormResponseSchema(TEST_FORM_ID, TEST_NAME, TEST_COMPONENTS, list,TEST_PROPERTIES,TEST_TYPE_FORM, TEST_VERSION,IS_DEFAULT_VALUE, TEST_CREATED_BY_ID,
+                String.valueOf(TEST_CREATED_ON), TEST_UPDATED_BY_ID, String.valueOf(TEST_UPDATED_ON));
+        Mockito.when(mockFormService.getRuntimeFormById(formDataSchemaTest.getFormId())).thenReturn(formResponseSchemaTest);
+        Mockito.when(mockIdGeneratorImpl.nextId()).thenReturn(BigInteger.valueOf(Long.parseLong(TEST_FORM_ID)));
+        Mockito.when(mockMongoTemplate.collectionExists(TP_RUNTIME_FORM_DATA +formDataSchemaTest.getFormId())).thenReturn(false);
+        Map<String, Object> formDataMap = new HashMap<>();
+        formDataMap.put(UNDERSCORE_ID,Long.parseLong(UNDERSCORE_ID_VALUE));
+        formDataMap.put(FORM_ID,TEST_FORM_ID);
+        formDataMap.put(VERSION, String.valueOf(1));
+        formDataMap.put(FORM_META_DATA, formDataSchemaTest.getFormMetaData());
+        formDataMap.put(FORM_DATA, formDataSchemaTest.getFormData());
+        formDataMap.put(CREATED_ON, Date.from(Instant.now()));
+        formDataMap.put(CREATED_BY_ID,CREATED_BY_USER_ID);
+        formDataMap.put(CREATED_BY_NAME, CREATED_BY_USER_NAME);
+        formDataMap.put(UPDATED_ON,TEST_UPDATED_ON);
+        formDataMap.put(UPDATED_BY_ID,UPDATED_BY_USER_ID);
+        formDataMap.put(UPDATED_BY_NAME,UPDATED_BY_USER_NAME);
+        Mockito.when(mockMongoTemplate.save(any(),anyString())).thenThrow(new MongoException("Duplicate key"));
+        Mockito.when(mockObjectMapper.convertValue(any(),eq(FormDataDefinition.class))).thenReturn(new FormDataDefinition());
+        Assertions.assertThrows(InvalidInputException.class,()->mockFormDataServiceImpl.saveFormData(formDataSchemaTest));
+    }
+
+    @Test
+    void saveFormDataWithUniqueKeyCheck1() throws IOException
```

Comment:
provide a name that explains hte test case

Reply:
I have changed the name